File: library/src/main/java/com/proxerme/library/api/DefaultLoginTokenManager.java
package com.proxerme.library.api;

import org.jetbrains.annotations.Nullable;

/**
 * TODO: Describe class
 *
 * @author Ruben Gees
 */
final class DefaultLoginTokenManager implements LoginTokenManager {

    private String currentToken;

    @Override
    @Nullable
    public String provide() {
        return currentToken;
    }

    @Override
    public void persist(@Nullable final String loginToken) {
        currentToken = loginToken;
    }
}

File: library/src/main/kotlin/me/proxer/library/internal/DefaultLoginTokenManager.kt
package me.proxer.library.internal

import me.proxer.library.LoginTokenManager

/**
 * @author Ruben Gees
 */
internal class DefaultLoginTokenManager : LoginTokenManager {

    @Volatile
    private var currentToken: String? = null

    override fun provide(): String? {
        return currentToken
    }

    override fun persist(loginToken: String?) {
        currentToken = loginToken
    }
}
